### PR TITLE
LG-3048: use case insensitive seach when adding emails to a team

### PR DESCRIPTION
**Pull request by solipet**
**Why?** Adding an existing user to the dashboard does not work with uppercase characters.

### CODE REVIEW

**Comment by solipet** [comment]
To be fair, the issue says it expects a warning that the user already exists, where this solution silently ignores the fact that you tried to add an existing user. If that's a problem, I can keep working on it.

**Comment by jmhooper** [comment]
@solipet: I don't think that is a problem. We can stay with the existing behavior for the bug fix.

File: app/forms/manage_users_form.rb [hunk 1]
```diff
@@ -35,4 +33,13 @@ def user_emails_are_valid_email_addresses
       errors.add(:base, "#{email} is not a valid email address")
     end
   end
+
+  def existing_users
+    User.where('lower(email) IN (?)', user_emails.map(&:downcase)).to_a
```
Comment (jmhooper) [inline app/forms/manage_users_form.rb:38]: Messed up my review comment.

One think we could probably do here is downcase `user_emails` on submit:

```ruby
def submit(user_emails)
  @user_emails = user_emails.map(&:downcase)
end
```

I'll leave whether to do that up to you.

Comment (zachmargolis) [reply app/forms/manage_users_form.rb:38]: What if we migrated the table to be `citext`? https://www.postgresql.org/docs/9.1/citext.html

Comment (solipet) [reply app/forms/manage_users_form.rb]: Yes! That cleans it up considerably. I wrote this prior to discovering that devise was already downcasing emails on save.

Comment (solipet) [reply app/forms/manage_users_form.rb]: @zachmargolis with devise already downcasing the emails, I don't want to mess with the existing schema/index. I tried to play with it, but couldn't quickly figure out how to get a rails migration to recognize the `citext` type.

Comment (zachmargolis) [reply app/forms/manage_users_form.rb]: Ok as long as Devise is doing essentially the same thing, it's good then

**Pull request merged**